t/op/magic.t

```diff
@@ -465,16 +465,22 @@ EOP
   is ($0, $char_with_plain_pv, 'compare $0 to non-UTF8-flagged');
 
   my $linux_cmdline_cr = sub {
-    open my $rfh, '<', "/proc/$$/cmdline" or skip "open: $!", 1;
-    return do { local $/; <$rfh> };
+    my $skip = shift // 1;
+    open my $rfh, '<', "/proc/$$/cmdline"
+      or skip "failed to read '/proc/$$/cmdline': $!", $skip;
+    my $got = do { local $/; <$rfh> };
+    $got=~s/\0\z//;
```

Comment:
FYI I'd probably write this as

```perl
return do { local $/; <$rfh> } =~ s/\0\z//r;
```

these days. It's a core perl unit test so we can rely on having 5.14+ features now. ;)

Reply:
Old habits die hard. :-)